lib/archethic/db/embedded_impl/inputs_reader.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Archethic.DB.EmbeddedImpl.InputsReader do
+  @moduledoc """
+  Inputs are stored by destination address. 1 file per address per ledger
+  """
+
+  alias Archethic.DB.EmbeddedImpl.InputsWriter
+  alias Archethic.TransactionChain.VersionedTransactionInput
+  alias Archethic.Utils
+
+  @spec get_inputs(ledger :: InputsWriter.ledger(), address :: binary()) ::
+          list(VersionedTransactionInput.t())
+  def get_inputs(ledger, address) do
+    filename = InputsWriter.address_to_filename(ledger, address)
+
+    case File.open(filename, [:read, :binary]) do
```

Comment:
Here you can use directly use `File.read` instead of `File.open` and `IO.binread`